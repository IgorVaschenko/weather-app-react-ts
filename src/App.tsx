import { useSelector } from 'react-redux'

import { RootState } from "store";
import InsideBlock from "components/InsideBlock";
import { getDateComponents } from "helpers/getDateComponents";
import { timeZone } from 'helpers/timeZone';

import GlobalStyles from "theme/globalStyles";

const App = () => {

  const timezone = useSelector((state: RootState) => state.weather.data?.city?.timezone) || 0
  const timezoneBit = useSelector((state: RootState) => typeof(state.weather.data?.timezone) !== undefined ? state.weather.data?.timezone : 0) 
  
  const timeIndex = Math.ceil(getDateComponents(timezone).hours / 3)
  const timezoneSource = timezone ? timezone : timeZone(timezoneBit) * 3600   

  const idWeatherOpen = useSelector((state: RootState) => state.weather.data?.list ? state.weather.data?.list[timeIndex - 1].weather[0].id : null) || 0
  const idWeatherBit = useSelector((state: RootState) => state.weather.data?.data ? state.weather.data?.data[0].weather.code : null)

  const actualBgWeather = idWeatherOpen ? idWeatherOpen : idWeatherBit || 0
  
  return (
    <>
      <InsideBlock idWeather={actualBgWeather} timesOfDay={getDateComponents(timezoneSource).hours}/>
      <GlobalStyles />
    </>
  );
}

export default App;
