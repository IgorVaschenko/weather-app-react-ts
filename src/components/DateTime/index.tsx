import Date from 'containers/Date';
import Time from 'containers/Time';

import { DateTimeBlock } from 'components/DateTime/components';

const DateTime = () => {

    return (
        <DateTimeBlock>
            <Time />
            <Date />
        </DateTimeBlock>
    );
}

export default DateTime;