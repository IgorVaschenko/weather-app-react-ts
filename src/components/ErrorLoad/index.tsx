import { FC } from 'react'
import { useDispatch } from 'react-redux';

import errorIcon from 'assets/error.svg'
import { setError } from 'store/actions/weatherActions';
import { ErrorWrap, ErrorImg, ErrorText } from 'components/ErrorLoad/components';

interface ErrorProps {
  message: string;
}

const ErrorLoad: FC<ErrorProps> = ({ message }) => {

  const dispatch = useDispatch()
  const handlerError = () => dispatch(setError())

  return (
    <ErrorWrap onClick={handlerError}>
      <ErrorImg src={errorIcon} />
      <ErrorText>Something went wrong...</ErrorText>
      <ErrorText>{message}</ErrorText>
    </ErrorWrap>
  );
};

export default ErrorLoad;
