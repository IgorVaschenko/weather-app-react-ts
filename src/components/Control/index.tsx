import { useSelector } from 'react-redux';

import { RootState } from 'store';
import SettingsInfo from 'components/SettingsInfo';
import Todo from 'components/Todo';
import Loader from 'components/Loader';
import ErrorLoad from 'components/ErrorLoad';

import { UpBlock } from 'components/Control/components'

const Control = () => {
    const loading = useSelector((state: RootState) => state.weather.loading)
    const error = useSelector((state: RootState) => state.weather.error)
    return (
        <UpBlock>
            <SettingsInfo />
            {loading ? <Loader /> : error ? <ErrorLoad message={error} /> : <Todo />}
        </UpBlock>
    );
}

export default Control;