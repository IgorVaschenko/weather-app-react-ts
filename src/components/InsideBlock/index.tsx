import { FC } from 'react'

import Control from 'containers/Control';
import WeekWeather from 'components/WeekWeather';
import SignIn from 'containers/SignIn';

import { InsideBlockWrap, OutsideBlock } from 'components/InsideBlock/components';

const InsideBlock: FC = () => {
    return (
        <OutsideBlock>
            <InsideBlockWrap>
                <SignIn />
                <Control />
                <WeekWeather />
            </InsideBlockWrap>
        </OutsideBlock>
    );
}

export default InsideBlock;