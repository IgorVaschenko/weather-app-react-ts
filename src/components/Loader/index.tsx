import { Loader, LoaderWrap } from 'components/Loader/components'

const LOADER_SIZE = 100

export default () => {
  return (
    <LoaderWrap>
      <Loader
        height={LOADER_SIZE}
        width={LOADER_SIZE}
        color="#AC5843"
        ariaLabel="loading"
        secondaryColor='gray'
      />
    </LoaderWrap>
  )
}
