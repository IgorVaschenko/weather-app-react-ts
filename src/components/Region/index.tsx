import { useSelector } from 'react-redux'
import { RootState } from 'store'
import { countries } from 'constants/countries'

import { RegionBlock, City, Country } from 'components/Region/components'

const Region = () => {
    const city = useSelector((state: RootState) => state.weather.data?.city?.name)
    const cityBit = useSelector((state: RootState) => state.weather.data?.city_name)

    const country = useSelector((state: RootState) => state.weather.data?.city?.country) || ''
    const countryBit = useSelector((state: RootState) => state.weather.data?.country_code) || ''

    const countryCode = country ? country : countryBit
    const countryName = city ? city : cityBit

    return (
        <RegionBlock>
            <City>{countryName}</City>
            <Country>{countries[countryCode]}</Country>
        </RegionBlock>
    );
}

export default Region;