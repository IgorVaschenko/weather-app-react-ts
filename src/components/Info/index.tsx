import DateTime from 'components/DateTime';
import Search from 'components/Search/Search';
import Region from 'components/Region';

import { WrapperRegion } from 'components/Info/components'

const Info = () => {
    return (
        <WrapperRegion>
            <DateTime />
            <Search />
            <Region />
        </WrapperRegion>
    );
}

export default Info;