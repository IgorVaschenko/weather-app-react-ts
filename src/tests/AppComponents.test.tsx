import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux';

import store from 'store';
import DateTime from 'components/DateTime';
import Info from 'components/Info';
import InsideBlock from 'components/InsideBlock';
import Loader from 'components/Loader';
import SettingsInfo from 'components/SettingsInfo';
import Toggle from 'components/Toggle';
import WeekWeather from 'components/WeekWeather';
import Control from 'containers/Control';
import Date from 'containers/Date';
import ErrorLoad from 'containers/ErrorLoad';
import OtherDay from 'containers/OtherDay';
import Region from 'containers/Region';
import Search from 'containers/Search/Search';
import SignIn from 'containers/SignIn';
import Time from 'containers/Time';
import Today from 'containers/Today';
import Todo from 'containers/Todo';
import Toggles from 'containers/Toggles';
import App from 'App'


describe('App Component', () => {
    it('Should render DateTime', () => {
        const component = render(
            <Provider store={store}>
                <DateTime />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render Info', () => {
        const component = render(
            <Provider store={store}>
                <Info />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render InsideBlock', () => {
        const component = render(
            <Provider store={store}>
                <InsideBlock />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render Loader', () => {
        const component = render(
            <Provider store={store}>
                <Loader />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render SettingsInfo', () => {
        const component = render(
            <Provider store={store}>
                <SettingsInfo />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render Toggle', () => {
        const component = render(
            <Provider store={store}>
                <Toggle
                    nameLeft={'url1'}
                    nameRight={'url2'}
                    pointer={true}
                    onClick={() => { }}
                />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render WeekWeather', () => {
        const component = render(
            <Provider store={store}>
                <WeekWeather />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render WeekWeather', () => {
        const component = render(
            <Provider store={store}>
                <Control />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render Date', () => {
        const component = render(
            <Provider store={store}>
                <Date />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render ErrorLoad', () => {
        const component = render(
            <Provider store={store}>
                <ErrorLoad message='error message' />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render OtherDay', () => {
        const component = render(
            <Provider store={store}>
                <OtherDay index={1} key={1} />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render Region', () => {
        const component = render(
            <Provider store={store}>
                <Region />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render Search', () => {
        const component = render(
            <Provider store={store}>
                <Search />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render SignIn', () => {
        const component = render(
            <Provider store={store}>
                <SignIn />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render Time', () => {
        const component = render(
            <Provider store={store}>
                <Time />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render Today', () => {
        const component = render(
            <Provider store={store}>
                <Today />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render Todo', () => {
        const component = render(
            <Provider store={store}>
                <Todo />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render Toggles', () => {
        const component = render(
            <Provider store={store}>
                <Toggles />
            </Provider>
        )
        expect(component.container).toBeInTheDocument()
    })
    it('Should render correctly', () => {
        const app = render(
            <Provider store={store}>
                <App />
            </Provider>
        )
        expect(app.container).toMatchSnapshot()
    })
})