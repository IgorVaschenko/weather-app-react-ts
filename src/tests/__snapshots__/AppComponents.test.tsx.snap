// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should render correctly 1`] = `
<div>
  <div
    class="sc-gFGZVQ jCxsYF"
  >
    <div
      class="sc-lbOyJj bQjjHE"
    >
      <button
        class="sc-ehmTmK btIDGY"
      >
        LOG OUT
      </button>
      <div
        class="sc-jOrMOR jFTBpo"
      >
        <div
          class="sc-dIouRR gTkoDI"
        >
          <div
            class="sc-idiyUo auRVj"
          >
            <div
              class="sc-evZas iPLDI"
            >
              <div
                class="sc-ksZaOG clqMBY"
              >
                Openweather
              </div>
              <div
                class="sc-breuTD cTWReb"
              >
                <label
                  class="sc-fnykZs htfJrF"
                >
                  <input
                    class="sc-bjUoiL ToSVL"
                    type="checkbox"
                  />
                  <span
                    class="sc-fEOsli iEUegb"
                  />
                </label>
              </div>
              <div
                class="sc-hAZoDl eGGtyQ"
              >
                WeatherBit
              </div>
            </div>
            <div
              class="sc-evZas iPLDI"
            >
              <div
                class="sc-ksZaOG clqMBY"
              >
                C °
              </div>
              <div
                class="sc-breuTD cTWReb"
              >
                <label
                  class="sc-fnykZs htfJrF"
                >
                  <input
                    class="sc-bjUoiL ToSVL"
                    type="checkbox"
                  />
                  <span
                    class="sc-fEOsli iEUegb"
                  />
                </label>
              </div>
              <div
                class="sc-hAZoDl eGGtyQ"
              >
                F °
              </div>
            </div>
          </div>
          <div
            class="sc-iqcoie lloXUC"
          >
            <div
              class="sc-jSMfEi cpQpzT"
            >
              <div
                class="sc-hKMtZM bcrKGA"
              >
                <div
                  class="sc-eCYdqJ gafvgz"
                >
                  03:52
                </div>
              </div>
              <div
                class="sc-bczRLJ euAnxl"
              >
                <div
                  class="sc-gsnTZi LYEDa"
                >
                  Monday
                   
                </div>
                <div
                  class="sc-dkzDqf RAfhd"
                >
                  26 September 2022
                </div>
              </div>
            </div>
            <form
              class="sc-gKXOVf VqfqD"
            >
              <input
                class="sc-iBkjds eeMINy"
                placeholder="enter city"
                value=""
              />
              <button
                class="sc-ftvSup lmWQjq"
              >
                Search
              </button>
            </form>
            <div
              class="sc-papXJ gAYypr"
            >
              <div
                class="sc-jqUVSM fHFsML"
              />
              <div
                class="sc-kDDrLX VzXzM"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-iTONeN JdmbY"
        >
          <div
            aria-busy="true"
            aria-label="mutating-dots-loading"
            data-testid="mutating-dots-loading"
            role="status"
            style="display: flex;"
          >
            <svg
              data-testid="mutating-dots-svg"
              height="150"
              id="goo-loader"
              width="150"
            >
              <filter
                id="fancy-goo"
              >
                <fegaussianblur
                  in="SourceGraphic"
                  result="blur"
                  stdDeviation="6"
                />
                <fecolormatrix
                  in="blur"
                  mode="matrix"
                  result="goo"
                  values="1 0 0 0 0  0 1 0 0 0  0 0 1 0 0  0 0 0 19 -9"
                />
                <fecomposite
                  in="SourceGraphic"
                  in2="goo"
                  operator="atop"
                />
              </filter>
              <g
                filter="url(#fancy-goo)"
              >
                <animatetransform
                  attributeName="transform"
                  attributeType="XML"
                  dur="1.2s"
                  from="0 50 50"
                  id="mainAnim"
                  repeatCount="indefinite"
                  to="359 50 50"
                  type="rotate"
                />
                <circle
                  cx="50%"
                  cy="40"
                  fill="#AC5843"
                  r="12.5"
                >
                  <animate
                    attributeName="cy"
                    attributeType="XML"
                    begin="0;cAnim1.end+0.2s"
                    calcMode="spline"
                    dur="0.6s"
                    id="cAnim1"
                    keySplines="0.09, 0.45, 0.16, 1;0.09, 0.45, 0.16, 1"
                    keyTimes="0;0.3;1"
                    values="40;20;40"
                  />
                </circle>
                <circle
                  cx="50%"
                  cy="60"
                  fill="#858585"
                  r="12.5"
                >
                  <animate
                    attributeName="cy"
                    attributeType="XML"
                    begin="0.4s;cAnim2.end+0.2s"
                    calcMode="spline"
                    dur="0.6s"
                    id="cAnim2"
                    keySplines="0.09, 0.45, 0.16, 1;0.09, 0.45, 0.16, 1"
                    keyTimes="0;0.3;1"
                    values="60;80;60"
                  />
                </circle>
              </g>
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-iNWwEs bjrfON"
      >
        <div
          class="sc-jfmDQi jqOSuF"
        >
          <div
            class="sc-kgUAyh fbqPdt"
          >
            <div
              class="sc-hTtwUo jZDLpF"
            />
            <div
              class="sc-lbxAil drqebZ"
            >
              <div
                class="sc-gSAPjG jGzqRr"
              >
                TODAY
              </div>
              <div
                class="sc-jgbSNz LHnnX"
              >
                0
                °
              </div>
            </div>
          </div>
          <div
            class="sc-dPyBCJ hJdjTS"
          >
            <div
              class="sc-BeQoi leeoRb"
            >
              TUE
            </div>
            <div
              class="sc-bBXxYQ ixnERY"
            />
            <div
              class="sc-cOFTSb ivtCVF"
            >
              0
              °
            </div>
          </div>
          <div
            class="sc-dPyBCJ hJdjTS"
          >
            <div
              class="sc-BeQoi leeoRb"
            >
              WED
            </div>
            <div
              class="sc-bBXxYQ ixnERY"
            />
            <div
              class="sc-cOFTSb ivtCVF"
            >
              0
              °
            </div>
          </div>
          <div
            class="sc-dPyBCJ hJdjTS"
          >
            <div
              class="sc-BeQoi leeoRb"
            >
              THU
            </div>
            <div
              class="sc-bBXxYQ ixnERY"
            />
            <div
              class="sc-cOFTSb ivtCVF"
            >
              0
              °
            </div>
          </div>
          <div
            class="sc-dPyBCJ hJdjTS"
          >
            <div
              class="sc-BeQoi leeoRb"
            >
              FRI
            </div>
            <div
              class="sc-bBXxYQ ixnERY"
            />
            <div
              class="sc-cOFTSb ivtCVF"
            >
              0
              °
            </div>
          </div>
          <div
            class="sc-dPyBCJ hJdjTS"
          >
            <div
              class="sc-BeQoi leeoRb"
            >
              SAT
            </div>
            <div
              class="sc-bBXxYQ ixnERY"
            />
            <div
              class="sc-cOFTSb ivtCVF"
            >
              0
              °
            </div>
          </div>
          <div
            class="sc-dPyBCJ hJdjTS"
          >
            <div
              class="sc-BeQoi leeoRb"
            >
              SUN
            </div>
            <div
              class="sc-bBXxYQ ixnERY"
            />
            <div
              class="sc-cOFTSb ivtCVF"
            >
              0
              °
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
